File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/collection/CollectionPaths.java
package dev.reuss.tmdb.domain.collection;


import dev.reuss.tmdb.value.id.CollectionId;

final class CollectionPaths {

    private static final String COLLECTION = "/collection";

    private CollectionPaths() {
    }

    static String details(CollectionId collectionId) {
        return collection(collectionId);
    }

    static String images(CollectionId collectionId) {
        return collection(collectionId) + "/images";
    }

    static String translations(CollectionId collectionId) {
        return collection(collectionId) + "/translations";
    }

    private static String collection(CollectionId collectionId) {
        return COLLECTION + "/" + collectionId.asString();
    }

}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/collection/CollectionPaths.kt
package dev.reuss.tmdb.domain.collection

import dev.reuss.tmdb.core.path.tmdbPath
import dev.reuss.tmdb.value.id.CollectionId

internal object CollectionPaths {
    fun details(collectionId: CollectionId): String = tmdbPath("collection", collectionId)

    fun images(collectionId: CollectionId): String = tmdbPath("collection", collectionId, "images")

    fun translations(collectionId: CollectionId): String = tmdbPath("collection", collectionId, "translations")
}
